src/nodes.coffee

```diff
@@ -1605,6 +1618,13 @@ exports.Code = class Code extends Base
     @bound       = tag is 'boundfunc'
     @isGenerator = !!@body.contains (node) ->
       (node instanceof Op and node.isYield()) or node instanceof YieldReturn
+    @isAsync     = !!@body.contains (node) ->
+      (node instanceof Op and node.isAwait()) or node instanceof AwaitReturn
+
+    @isContinuation = @isGenerator or @isAsync
+
+    if @isAsync and @isGenerator
+      @error "function can't contain both yield and await"
```

Comment:
`@operator.error` should give a nicer error message. A test for that would be nice, too!